src/app/_components/BlogSearchInput.tsx

```diff
@@ -2,30 +2,48 @@
 
 import { useState } from 'react'
 
+import { useSearchParams } from 'next/navigation'
+
 import { useDebounceCallback } from 'usehooks-ts'
 
+import { useSearch } from '@/hooks/useSearch'
 import { useUpdateSearchParams } from '@/hooks/useUpdateSearchParams'
 
 import { SearchInput } from '@/components/SearchInput'
 
+import { PAGE_KEY, SEARCH_KEY } from '@/constants/searchParams'
+
 type BlogSearchInputProps = {
-  searchQuery: string
+  query: ReturnType<typeof useSearch>['searchQuery']
 }
 
-export function BlogSearchInput({ searchQuery }: BlogSearchInputProps) {
-  const [value, setValue] = useState(searchQuery)
+const TYPE_DEBOUNCE_DELAY = 400
+
+export function BlogSearchInput({ query }: BlogSearchInputProps) {
+  const [value, setValue] = useState(query)
+  const params = useSearchParams()
 
   const updateSearchParams = useUpdateSearchParams()
 
   const debouncedUpdateSearchParams = useDebounceCallback(
     updateSearchParams,
-    500,
+    TYPE_DEBOUNCE_DELAY,
   )
 
   function handleSearchChange(event: React.ChangeEvent<HTMLInputElement>) {
     const newValue = event.target.value
     setValue(newValue)
-    debouncedUpdateSearchParams({ search: newValue, page: '1' })
+
+    const existingPageParam = params.get(PAGE_KEY)
+
+    if (existingPageParam && existingPageParam !== '1') {
+      debouncedUpdateSearchParams({
+        [SEARCH_KEY]: newValue,
+        [PAGE_KEY]: '1',
+      })
+    } else {
+      debouncedUpdateSearchParams({ [SEARCH_KEY]: newValue })
+    }
   }
```

Comment:
I'm guessing we're going to want to refactor this into a custom `useDebounce` hook? Not a blocker.

Reply:
How would you want to refactor this?

Reply:
Ah, I think I falsely assumed we were going to need something similar for sort and categories. But we're not going to need to debounce those. 🙏🏼

Reply:
I see :) We just need debounce for search and pagination.